### ISSUE TITLE
prop-types complains about stateless functions

### ISSUE THREAD

**Issue by elijahsmith** [yannickcr/eslint-plugin-react#302]
there is no way to define propTypes in stateless function component types, but the ```react/prop-types``` rule still errors.

**Comment by yannickcr** [comment]
Actually you can totally define `propTypes` in stateless function components.
```js
function Hello(props) {
  return <div>Hello {props.name}</div>;
}
Hello.propTypes = {
  name: React.PropTypes.string.isRequired
}
```
You can also define `defaultProps` and use `context`. 
A good article about stateless components: [Functional Stateless Components in React 0.14](https://medium.com/@joshblack/stateless-components-in-react-0-14-f9798f8b992d)

**Comment by gpbmike** [comment]
This is exactly what I was looking for thanks.
